### PR TITLE
[Jetpack Content Migration Flow] Add tracking events

### LINKED ISSUE #17572: Android - Unify Tracks data
This task is to add tracks events to the migration flow.

The following events should be tracked in the WordPress to Jetpack Migration Flow*:

### Tracks Events
- `jpmigration_welcome_screen_shown`
   - Fired when the Welcome screen of the migration flow is shown
- `jpmigration_welcome_screen_continue_button_tapped`
   - Fired when the `Continue` button is tapped on the Welcome screen
- `jpmigration_welcome_screen_help_button_tapped`
   - Fired when the `Need Help` button is tapped on the Welcome Screen
- `jpmigration_welcome_screen_avatar_tapped`
   - Fired when the User avatar icon is tapped on the Welcome Screen
- `jpmigration_notifications_screen_shown`
   - Fired when the Notifications screen of the migration flow is shown
- `jpmigration_notifications_screen_continue_button_tapped`
   - Fired when the `Continue` button is tapped on the Notifications screen
- `jpmigration_thanks_screen_shown`
   - Fired when the Success screen of the migration flow is shown
- `jpmigration_thanks_screen_finish_button_tapped`
   - Fired when the `Finish` button is tapped on the Success screen
- `jpmigration_please_delete_wordpress_card_tapped`
   - Fired when the "Please delete WordPress app" card is tapped on My Site
- `jpmigration_please_delete_wordpress_screen_shown`
   - Fired when the "Please delete WordPress app" screen is shown
- `jpmigration_please_delete_wordpress_gotit_tapped`
   - Fired when the `Got It` button is tapped on the "Please delete WordPress app" screen
- `jpmigration_please_delete_wordpress_help_button_tapped`
   - Fired when the `Need Help` button is tapped on the "Please delete WordPress app" screen
- `jpmigration_error_screen_shown`
   - Fired when the Error screen is shown in the migration flow
- `jpmigration_error_screen_help_button_tapped`
   - Fired when the `Need Help` button is tapped on the Error screen
- `jpmigration_error_screen_retry_button_tapped`
   - Fired when the `Try Again` button is tapped on the Error screen

TODO on the issue:

- [x] add info when the tracking event should be fired
- [x] add properties if necessary

Related:
- https://github.com/wordpress-mobile/WordPress-Android/issues/17574

---

/* Subject to change for completeness or cleanup.

### LINKED ISSUE #17574: Android - Track data for WordPress app compatibility
## Description
We should track if the Jetpack app detects an compatible version of the WordPress app.

Event: `jpmigration_wordpressapp_detected`
Property: `compatible` (values: `true`, `false`)

### PR DESCRIPTION
**Pull request by ovitrif**
Fixes #17572
Fixes #17574

This PR adds tracking for the Migration Flow in the Jetpack App, and unifies tracking via `ContentMigrationAnalyticsTracker`.

It tackles both issue linked above.

To test:

#### 1️⃣ Test migration happy flow tracking events

1. Build & install the WordPress app `wordpressWasabiDebug` flavor, open app and login
2. Install the Jetpack app of the same flavor (e.g. `jetpackWasabiDebug`)
3. Open the Jetpack app
4. The Welcome Screen of the Jetpack migration flow should be shown
   **Verify** in logcat that the following tracking events are logged:
   1. `jpmigration_wordpressapp_detected`
   ```
   🔵 Tracked: jpmigration_wordpressapp_detected, Properties: {"compatible":true}
   ```
   2. `jpmigration_welcome_screen_shown`
   ```
   🔵 Tracked: jpmigration_welcome_screen_shown
   ``` 
5. Tap `Need Help`
   - **Verify** the `jpmigration_welcome_screen_help_button_tapped` tracking event is logged:
   ```
   🔵 Tracked: jpmigration_welcome_screen_help_button_tapped
   ```
6. Go back
7. Tap on the user avatar image in the top right corner
   **Verify** the `jpmigration_welcome_screen_avatar_tapped` is tracked
8. Go back
9. Tap the `Continue` button on the Welcome Screen
10. The Notifications screen of the flow should be shown
    **Verify** the following events are tracked:
    1. `jpmigration_welcome_screen_continue_button_tapped`
    2. `jpmigration_notifications_screen_shown`
11. Tap the `Continue` button on the Notifications screen
12. The success/thanks screen of the flow should be shown
    **Verify** the following events are tracked:
    1. `jpmigration_notifications_screen_continue_button_tapped`
    2. `jpmigration_thanks_screen_shown`
13. Tap the `Finish` button on the success/thanks screen
14. The `My Site` screen should be shown
    **Verify** the `jpmigration_thanks_screen_finish_button_tapped` is tracked
15. On the `My Site` screen tap the `Please delete the WordPress app` card
16. The "You no longer need the WordPress app" screen should be shown
    **Verify** the following events are tracked:
    1. `jpmigration_please_delete_wordpress_card_tapped`
    2. `jpmigration_please_delete_wordpress_screen_shown`
17. Tap `Need Help` on the screen
    **Verify** the `jpmigration_please_delete_wordpress_help_button_tapped` event is tracked
18. Go back
19. Tap `Got it` on the screen to delete WordPress app
20. The `My Site` screen should be shown
    **Verify** the `jpmigration_please_delete_wordpress_gotit_tapped` event is tracked

<details>
<summary><h4>2️⃣ Test incompatible WordPress app detected</h4></summary>

1. Prerequisite: WordPress app should be installed and logged in
2. Uninstall Jetpack App if installed
3. In `JetpackAppMigrationFlowUtils` change `minimumSupportedVersion` to `21.4`
   > This is a hack to mimic real-life scenarios where the Jetpack app requires a higher version of the WordPress than what's currently installed.
4. Build and install the Jetpack app of the same flavor that the installed WordPress app is (e.g. `wasabiDebug`)
5. Open the Jetpack App
6. The migration flow should **not** be shown, instead the Landing screen should appear
   **Verify** the `jpmigration_wordpressapp_detected` event is tracked with the "compatible" property false
   ```
   🔵 Tracked: jpmigration_wordpressapp_detected, Properties: {"compatible":false}
   ```
</details>

<details>
<summary><h4>3️⃣ Test error screen events tracking in the migration flow</h4></summary>

1. Prerequisite: WordPress app should be installed and logged in
2. Uninstall Jetpack App if installed
3. In `build.gradle` change `JETPACK_LOCAL_USER_FLAGS` build config value to `false`:
   ```
   buildConfigField "boolean", "JETPACK_LOCAL_USER_FLAGS", "false"
   ```
   **Note** _This is a hardcode solution to make the migration flow always fail._
4. Sync project
5. Build and install the Jetpack app of the same flavor that the installed WordPress app is (e.g. `wasabiDebug`)
6. Open the Jetpack app
7. The Error screen of the migration flow should be shown
   **Verify** the `jpmigration_error_screen_shown` event is tracked
8. Tap the `Need Help` button
   **Verify** the `jpmigration_error_screen_help_button_tapped` is tracked
9. Go back
10. Tap the `Try Again` button
   **Verify** the `jpmigration_error_screen_retry_button_tapped` is tracked

</details>

## Regression Notes

1. Potential unintended areas of impact
   Jetpack content migration flow.

12. What I did to test those areas of impact (or what existing automated tests I relied on)
   Manual testing, & relied on current automated tests.

13. What automated tests I added (or what prevented me from doing so)
   Added tests for tracking in:
   - `JetpackAppMigrationFlowUtilsTest`
   - `MySiteViewModelTest`
   - `JetpackMigrationViewModelTest`

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/test/java/org/wordpress/android/ui/mysite/MySiteViewModelTest.kt [hunk 1]
```diff
@@ -2250,6 +2253,20 @@ class MySiteViewModelTest : BaseUnitTest() {
         assertThat((getDashboardTabLastItems()[0] as SingleActionCard).imageResource).isEqualTo(expected)
     }
 
+    @Test
+    fun `JP migration scuccess card click should be tracked`() {
```
Comment (mkevins) [inline WordPress/src/test/java/org/wordpress/android/ui/mysite/MySiteViewModelTest.kt]: NP: typo

**Review by mkevins — APPROVED** [review]
Thanks for these changes Ovi! The code looks good. Also I tested the listed scenarios on a Pixel 3a (physical device):

:heavy_check_mark: :one: Test migration happy flow tracking events
:heavy_check_mark: :two: Test incompatible WordPress app detected
:heavy_check_mark: :three: Test error screen events tracking in the migration flow

And all three scenarios are working mostly as described. I added a tiny nit-pick comment, not blocking (just a typo), and additionally, I observed that the welcome screen event was emitted on each update. I don't consider this blocking, since the PR adds a lot of value in it's current state. But, I've found a solution, and it is on this suggestion branch, and fast-forwardable if you'd like to include it in this PR before merging: https://github.com/wordpress-mobile/WordPress-Android/compare/task/jpmigration-track-events...task/jpmigration-track-events--suggestion-fix-duplicate-welcome-tracks

**Review by antonis — APPROVED** [review]
Great work @ovitrif 👍 
Everything LGTM and works as expected 🎉 

ps. I've tried the failing `JetpackFeatureRemovalOverlayUtilTest` test locally and it worked for me. I've restarted the test process 🤞 

> I don't consider this blocking, since the PR adds a lot of value in it's current state. But, I've found a solution, and it is on this suggestion branch, and fast-forwardable if you'd like to include it in this PR before merging: https://github.com/wordpress-mobile/WordPress-Android/compare/task/jpmigration-track-events...task/jpmigration-track-events--suggestion-fix-duplicate-welcome-tracks

@mkevins suggestion also makes sense to me 👍

File: WordPress/src/test/java/org/wordpress/android/ui/mysite/MySiteViewModelTest.kt (hunk 1 above)
Comment (ovitrif) [reply WordPress/src/test/java/org/wordpress/android/ui/mysite/MySiteViewModelTest.kt]: Thank you for noticing this typo @mkevins 🙇 

Fixed in: https://github.com/wordpress-mobile/WordPress-Android/pull/17585/commits/7c508969ff62f337d67f44a62b9579ad929a45d8

**Comment by ovitrif** [comment]
>I added a tiny nit-pick comment, not blocking (just a typo), and additionally, I observed that the welcome screen event was emitted on each update. I don't consider this blocking, since the PR adds a lot of value in it's current state. But, I've found a solution, and it is on this suggestion branch, and fast-forwardable if you'd like to include it in this PR before merging: [task/jpmigration-track-events...task/jpmigration-track-events--suggestion-fix-duplicate-welcome-tracks](https://github.com/wordpress-mobile/WordPress-Android/compare/task/jpmigration-track-events...task/jpmigration-track-events--suggestion-fix-duplicate-welcome-tracks)

Thank you for reporting and providing a fix for the issue with duplicate tracking of the Welcome screen event 🙇 

I included your suggestion in https://github.com/wordpress-mobile/WordPress-Android/pull/17585/commits/e0106406efc44a85dea4038eec3a2af19b516a63 🙏

Edit: The fix works as expected and there's no regression.

**Comment by antonis** [comment]
> Thank you for reporting and providing a fix for the issue with duplicate tracking of the Welcome screen event 🙇
> I included your suggestion in https://github.com/wordpress-mobile/WordPress-Android/commit/e0106406efc44a85dea4038eec3a2af19b516a63 🙏

Thank you both for your work on this 🙇 
As soon as the tests pass we could merge this and request a new beta that would also include the issue blocking testing (ref: p1669805217835989/1669726022.223989-slack-C0180B5PRJ4)

**Comment by ovitrif** [comment]
Seeing the CI checks passed on this PR and the fact that it's tested we can proceed to merge the PR.
@jkmassel Pinging you here to inform we need a new beta for Android:

> As soon as the tests pass we could merge this and request a new beta that would also include the issue blocking testing (ref: p1669805217835989/1669726022.223989-slack-C0180B5PRJ4)

**Pull request merged**